### PR TITLE
Add tests for Sabre Auth plugin + make getCurrentUser compatible

### LINKED ISSUE #19969: CardDAV cookie authentication
The webdav backend seems to have problems with cookies in the request header:

The following curl (from command line) works as expected:
```
$ curl testuser:testpasswd@localhost:8080/remote.php/carddav/addressbooks/testuser/contacts/15e3775c-87ae-4fa0-82c1-fa7fd845a4d4.vcf 
BEGIN:VCARD
VERSION:3.0
PRODID:-//ownCloud//NONSGML Contacts 0.5.0.0//EN
N:Leppelsack;Hendrik;;;
UID:15e3775c-87ae-4fa0-82c1-fa7fd845a4d4
REV:2015-10-21T19:09:28+00:00
FN:
END:VCARD
```

while the same curl with the session cookies copied from the browser returns a "not found"-exception:
```
$ curl --cookie "oc_sessionPassphrase=cwTfwcYDgpy80KkBgPMW73JTcLJWF5Lqs7OGEIn6JtBlmC6v26bvAcRpnCPnbvYVJPjh07xlSEZiTa4XWGv9Pyvh0uhUaF8TVI2z4o2kw1ZTbg%2FiVPRbwqehZCKzidDk; ocbjoy4fintd=8uso7q3ugst8md4572q2chklo3" testuser:testpasswd@localhost:8080/remote.php/carddav/addressbooks/testuser/contacts/15e3775c-87ae-4fa0-82c1-fa7fd845a4d4.vcf
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:sabredav-version>2.1.7</s:sabredav-version>
  <s:exception>Sabre\DAV\Exception\NotFound</s:exception>
  <s:message>Card not found</s:message>
</d:error>
```

This happens on owncloud stable8.2.
I can't test it on 9.0 as carddav is currently broken because of the dav changes in core.

cc @DeepDiver1975

### PR DESCRIPTION
**Pull request by LukasReschke**
Master version of https://github.com/owncloud/core/pull/20001, had to make the code non-static and make it use DI for the tests.

Fixes #19969

According to my local tests everything works still the same as before but that doesn't have to mean anything. :wink: 

@icewind1991 @DeepDiver1975 Thoughts? THX.

### CODE REVIEW

File: apps/dav/lib/connector/sabre/auth.php [hunk 1]
```diff
@@ -64,24 +81,21 @@ protected function isDavAuthenticated($username) {
...
 	 */
 	protected function validateUserPass($username, $password) {
-		if (\OC_User::isLoggedIn() &&
-			$this->isDavAuthenticated(\OC_User::getUser())
+		if ($this->userSession->isLoggedIn() &&
+			$this->isDavAuthenticated($this->userSession->getUser()->getUID())
 		) {
-			\OC_Util::setupFS(\OC_User::getUser());
-			\OC::$server->getSession()->close();
+			\OC_Util::setupFS($this->userSession->getUser()->getUID());
```
Comment (LukasReschke) [inline apps/dav/lib/connector/sabre/auth.php]: @icewind1991 Do we by the way have non static replacement for stuff like this?

**Comment by DeepDiver1975** [comment]
looks good

@nickvergessen can you please smash this? THX

**Comment by nickvergessen** [comment]
Smashing :hourglass_flowing_sand:

**Comment by nickvergessen** [comment]
Smashed: :white_check_mark:

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**